src/PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp

```diff
@@ -0,0 +1,139 @@
+ *
+ * \note Kerr's original "advanced Eddington-Finkelstein" form of his metric is
+ * retrieved by performing the additional transformation \f$t = u - r\f$.
+ *
+ * The Cartesian Kerr-Schild coordinates \f$(x, y, z)\f$ can then be defined via
+ *
+ * \f{align*}
+ * x &= (r \cos\phi + a \sin\phi)\sin\theta =
+ * \sqrt{r^2 + a^2}\sin\theta\cos\left(\phi - \tan^{-1}(a/ r)\right),\\
+ * y &= (r \sin\phi - a \cos\phi)\sin\theta =
+ * \sqrt{r^2 + a^2}\sin\theta\sin\left(\phi - \tan^{-1}(a/ r)\right),\\
+ * z &= r \cos\theta.
+ * \f}
+ *
+ * These are the coordinates Solutions::KerrSchild is written in.
+ * Using the identities
+ *
+ * \f{align*}
+ * r^2 &= \frac{x^2 + y^2 + z^2 - a^2}{2} +
+ * \sqrt{\left(\frac{x^2 + y^2 + z^2 - a^2}{2}\right)^2 + a^2 z^2},\\
+ * \sin\theta &= \sqrt{\frac{x^2 + y^2}{r^2 + a^2}},\quad
+ * \cos\theta = \frac{z}{r},\\
+ * \sin\phi &= \frac{x a + y r}{\sqrt{(x^2 + y^2)(r^2 + a^2)}},\quad
+ * \cos\phi = \frac{x r - y a}{\sqrt{(x^2 + y^2)(r^2 + a^2)}},
+ * \f}
+ *
+ * the Jacobian of the transformation to Cartesian Kerr-Schild coordinates is
+ *
+ * \f{align*}
+ * \dfrac{\partial(x, y, z)}{\partial(r, \theta, \phi)} &=
+ * \left(\begin{array}{ccc}
+ * \dfrac{xr - ya}{r^2 + a^2} &
+ * \dfrac{xz}{r}\sqrt{\dfrac{r^2 + a^2}{x^2 + y^2}} & -y\\
+ * \dfrac{xa + yr}{r^2 + a^2} &
+ * \dfrac{yz}{r}\sqrt{\dfrac{r^2 + a^2}{x^2 + y^2}} & x\\
+ * \dfrac{z}{r} & - r\sqrt{\dfrac{x^2 + y^2}{r^2 + a^2}} & 0
+ * \end{array}\right),
+ * \f}
+ *
+ * which can be used to transform tensors between both coordinate systems.
+ *
+ * \note Sometimes (especially in the GRMHD community), the generalized
+ * Eddington-Finkelstein coordinates as defined above are referred to as
+ * "spherical Kerr-Schild" coordinates. These should not be confused with the
+ * true spherical coordinates defined so that \f$r^2 = x^2 + y^2 + z^2\f$.
+ */
+class KerrSchildCoords {
+ public:
+  KerrSchildCoords() = default;
+  KerrSchildCoords(const KerrSchildCoords& /*rhs*/) = delete;
+  KerrSchildCoords& operator=(const KerrSchildCoords& /*rhs*/) = delete;
+  KerrSchildCoords(KerrSchildCoords&& /*rhs*/) noexcept = default;
+  KerrSchildCoords& operator=(KerrSchildCoords&& /*rhs*/) noexcept = default;
+  ~KerrSchildCoords() = default;
+
+  explicit KerrSchildCoords(double spin_a) noexcept;
+
+  /// Kerr-Schild \f$r^2\f$ in terms of the Cartesian coordinates.
+  template <typename DataType>
+  Scalar<DataType> r_coord_squared(
+      const tnsr::I<DataType, 3, Frame::Inertial>& cartesian_coords) const
+      noexcept;
+
+  /// The spatial components of the Jacobian of the transformation.
+  template <typename DataType>
+  tnsr::Ij<DataType, 3, Frame::Inertial> jacobian_matrix(
+      const tnsr::I<DataType, 3, Frame::Inertial>& cartesian_coords) const
+      noexcept;
+
+  /// Transforms a spatial vector from generalized Eddington-Finkelstein or
+  /// "spherical Kerr-Schild" coordinates to Cartesian coordinates.
+  // 'spherical_ks' as opposed to (true) 'spherical'
```

Comment:
Is this supposed to be part of the doxygen comment? It doesn't really tell me a whole lot and seems like it should be discussed in the main dox comment only.